shared-libs/transitions/src/transitions/update_clinics.js

```diff
@@ -1,107 +1,63 @@
 
-const associateContact = (doc, contact, callback) => {
-  const self = module.exports;
-
-  // reporting phone stayed the same and contact data is up to date
-  if (
-    doc.from === contact.phone &&
-    doc.contact &&
-    contact._id === doc.contact._id
-  ) {
-    return callback();
-  }
-
-  if (contact.phone !== doc.from) {
-    contact.phone = doc.from;
-    db.medic.put(contact, err => {
-      if (err) {
-        logger.error(`Error updating contact: ${JSON.stringify(err, null, 2)}`);
-        return callback(err);
-      }
-      self.setContact(doc, contact, callback);
-    });
-  } else {
-    self.setContact(doc, contact, callback);
-  }
-};
-
-const getHydratedContact = (id, callback) => {
-  return lineage
-    .fetchHydratedDoc(id)
-    .then(contact => {
-      callback(null, contact);
-    })
-    .catch(err => {
-      callback(err);
-    });
-};
-
-const getContact = (doc, callback) => {
+const getContact = doc => {
   if (doc.refid) {
-    // use reference id to find clinic if defined
-    let params = {
+    const params = {
       key: ['external', doc.refid],
       include_docs: true,
       limit: 1,
     };
-    db.medic.query(
-      'medic-client/contacts_by_reference',
-      params,
-      (err, data) => {
-        if (err) {
-          return callback(err);
-        }
+
+    return db.medic
+      .query('medic-client/contacts_by_reference', params)
+      .then(data => {
         if (!data.rows.length) {
-          return callback();
+          return;
         }
+
         const result = data.rows[0].doc;
         if (result.type === 'person') {
-          return getHydratedContact(result._id, callback);
-        }
-        if (result.type === 'clinic') {
+          return lineage.fetchHydratedDoc(result._id);
+        } else {
           const id = result.contact && result.contact._id;
           if (!id) {
-            return callback(null, result.contact || { parent: result });
+            return result.contact || { parent: result };
           }
 
-          return getHydratedContact(id, callback);
+          return lineage.fetchHydratedDoc(id);
         }
-        callback();
-      }
-    );
-  } else if (doc.from) {
-    let params = {
+      });
+  }
+
+  if (doc.from) {
+    const params = {
       key: String(doc.from),
       include_docs: false,
       limit: 1,
     };
-    db.medic.query('medic-client/contacts_by_phone', params, (err, data) => {
-      if (err) {
-        return callback(err);
-      }
-      if (data.rows.length && data.rows[0].id) {
-        return getHydratedContact(data.rows[0].id, callback);
-      }
-      return callback();
-    });
-  } else {
-    callback();
+
+    return db.medic
+      .query('medic-client/contacts_by_phone', params)
+      .then(data => {
+        if (!data.rows.length || !data.rows[0].id) {
+          return;
+        }
+
+        return lineage.fetchHydratedDoc(data.rows[0].id);
+      });
   }
+
+  return Promise.resolve();
```

Comment:
I find this quite hard to follow, I think because there's a lot going on in each branch of the `if`/`else`. Consider splitting each branch out as an individual function, eg:

```js
const getContact = doc => {
  if (doc.refid) {
    return getContactByRefId(...);
  }
  if (doc.from) {
    return getContactByPhone(...);
  }
  return Promise.resolve();
};
```

Reply:
Done!